### PR TITLE
Docs/guidelines

### PR DESCRIPTION
**Pull request by Danphillipz**
#### 📲 What

New CONTRIBUTING.md file

#### 🤔 Why

New contribution file to help new developers get started on the stacks project. 
Provides information about how the repository is set up and how to contribute to it

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

**Review by abdrah1 — COMMENTED** [review]
great document! 
was just wondering if it would be worth adding some info about the pipelines that are ran for every PR/commit.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+```
+
+This will set up [Husky](https://typicode.github.io/husky/) to manage commit quality which enforces naming and documentation conventions upon commits through [commitizen](https://commitizen-tools.github.io/commitizen/).
+
+## Naming Conventions and Coding Style
+
+When working with the Stacks project, adhere to the following naming conventions and coding style guidelines:
+
+- Ensure husky has been configured to enforce commit standards
```
Comment (abdrah1) [inline CONTRIBUTING.md:61]: how about just adding some brief info (with an example) of what the commit standards are?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+- *accessibility* generator: Use Git to review the differences between the `cypress-accessibility` branch and the `cypress-baseline` branch. Again creating the relevant unit tests.
+
+***Step 3: Implementation***
+
+It is worth reviewing existing plugins to understand how we utilise the `@nx/devkit` and `@ensono-stacks/core` to create and manipulate files.
+
+Some important initial checks:
+
+1. `verifyPluginCanBeInstalled()`: Checks to see if the workspace is compatible
```
Comment (abdrah1) [inline CONTRIBUTING.md:142]: is it worth mentioning that the these 2 functions (verifyPluginCanBeInstalled, hasGeneratorExecutedForProject) come from the @ensono-stacks/core package and should be called inside any generator for validation purposes.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+2. `hasGeneratorExecutedForProject()`: If the generator can only be ran once, this ensures that the `nx.json` and stacks `executedGenerators` list is correctly updated.
+
+Once the code has been written for your generators, you must create corresponding e2e tests to ensure that your plugin can be built, published and executed within a stacks/nx workspace.
+
+## Testing
+
+### Unit Testing
+
+Unit tests should looks to cover all areas of a generator in regards to file creation and manipulation. By following a TDD approach this should ensure maximum coverage for all of the files which are affected by a generator.
```
Comment (abdrah1) [inline CONTRIBUTING.md:152]:
```suggestion
Unit tests should look to cover all areas of a generator in regards to file creation and manipulation. By following a TDD approach this should ensure maximum coverage for all of the files which are affected by a generator.

```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+## Get Started
+
+### System Requirements
+
+To set up the Stacks workspace, ensure that you have the following installed on your system:
+
+- npm
+
+- Node.js version 18 or above
```
Comment (Elliot-Evans-95) [inline CONTRIBUTING.md:31]: I think this is fine but might be worth pointing to the exact version we are using, aka LTS (v18.16.1)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+The following are recommended plugins for visual studio code:
```
Comment (Elliot-Evans-95) [inline CONTRIBUTING.md:33]: Are there any notes for the Webstorm IDE? If there isn't much around the repo I can ask Cerden and George and see that we recommend as we all use Webstorm.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+To run your e2e tests you must use the e2e executor:
+
+```bash
+nx e2e <plugin name>-e2e
+```
+
+### Testing packages locally
+
+- Option 1: Use npm linking
+  - TO BE COMPLETED - @Metecan
```
Comment (Elliot-Evans-95) [inline CONTRIBUTING.md:199]: Are we ok to have this in the main branch without the notes about npm linking?

File: CONTRIBUTING.md (hunk 1 above)
Comment (Danphillipz) [reply CONTRIBUTING.md:61]: Updated, please review

File: CONTRIBUTING.md (hunk 2 above)
Comment (Danphillipz) [reply CONTRIBUTING.md:142]: Updated, please review

File: CONTRIBUTING.md (hunk 4 above)
Comment (Danphillipz) [reply CONTRIBUTING.md:31]: Updated, please review

File: CONTRIBUTING.md (hunk 5 above)
Comment (Danphillipz) [reply CONTRIBUTING.md:33]: Agreed to complete at a later date

File: CONTRIBUTING.md (hunk 6 above)
Comment (Danphillipz) [reply CONTRIBUTING.md:199]: Updated with linking guide

**Comment by Danphillipz** [comment]
> great document! was just wondering if it would be worth adding some info about the pipelines that are ran for every PR/commit.

Pipelines are documented in the "Releasing packages and publishing to NPM" section, happy with that?

**Review by abdrah1 — APPROVED** [review]

**Review by Elliot-Evans-95 — APPROVED** [review]

**Pull request merged**